src/components/newContact/NewContact.tsx

```diff
@@ -0,0 +1,95 @@
+import React, { FC, useState } from 'react'
+import styles from './newContact.module.scss'
+import { useDispatch, useSelector } from 'react-redux'
+import { selectIsLoading } from '../../redux/contacts/selectors'
+
+// Components
+import Header from '../header/Header'
+import { setNewContact } from '../../redux/contacts/actions'
+
+const NewContact: FC = (props) => {
+  const loading = useSelector(selectIsLoading);
+
+  const dispatch = useDispatch();
+  const [contactInfo, setContactInfo] = useState({
+    firstName: '',
+    lastName: '', 
+    phone: '',
+    email: '',
+    address1: '',
+    address2: '',
+    city: '',
+    state: '',
+    zip: ''
+  })
+
+  const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    e.persist();
+    setContactInfo((prevState) => ({
+      ...prevState,
+      [e.target.name]: e.target.value,
+    }));
+  };
+
+  const handleSave = () => {
+    const data = {
+      firstName: contactInfo.firstName,
+      lastName: contactInfo.lastName,
+      phone: contactInfo.phone,
+      email: contactInfo.email,
+      address1: contactInfo.address1,
+      address2: contactInfo.address2,
+      city: contactInfo.city,
+      state: contactInfo.state,
+      zip: contactInfo.zip
+    }
+    dispatch(setNewContact(data))
+    setContactInfo({
```

Comment:
If you add this into redux you can clear it only on success instead of always clearing it after dispatching

Reply:
I updated this to just use the state of the form, rather than the `useState` hook, however, I'm finding that my `newContact` is getting deleted when looking at the fulfilled step in the redux dev tool.